scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamOp.scala

```diff
@@ -51,9 +78,131 @@ object BeamOp extends Serializable {
+              .apply(GroupByKey.create[K, V]()).setCoder(KvCoder.of(OrderedSerializationCoder(ordK), IterableCoder.of(OrderedSerializationCoder(ordV))))
+              .apply(SortGroupedValues[K, V])
+
+            planMapGroup[K, V, U](groupedSortedValues, reduceFn)
+              .apply(ParDo.of(FlatMapFn[KV[K, java.lang.Iterable[U]], KV[K, U]]{ elem =>
+                elem.getValue.asScala.map(KV.of(elem.getKey, _))
+              })).setCoder(KvCoder.of(OrderedSerializationCoder(ordK), kryoCoder))
+              .apply(KVToTuple[K, U])
+          }
+        },
+        kryoCoder)
+    }
+    def sorted(implicit ordK: Ordering[K], ordV: Ordering[V], kryoCoder: KryoCoder): BeamOp[(K, V)] = {
+      TransformBeamOp[(K, V), (K, V)](
+        op,
+        new PTransform[PCollection[(K, V)], PCollection[(K, V)]]() {
+          override def expand(input: PCollection[(K, V)]): PCollection[(K, V)] = {
+            input
+              .apply(TupleToKV[K, V](OrderedSerializationCoder(ordK), OrderedSerializationCoder(ordV)))
+              .apply(GroupByKey.create[K, V]()).setCoder(KvCoder.of(OrderedSerializationCoder(ordK), IterableCoder.of(OrderedSerializationCoder(ordV))))
+              .apply(SortGroupedValues[K, V])
+              .apply(ParDo.of(FlatMapFn[KV[K, java.lang.Iterable[V]], KV[K, V]] { elem =>
+                elem.getValue.asScala.map(x => KV.of(elem.getKey, x))
+              })).setCoder(KvCoder.of(OrderedSerializationCoder(ordK), OrderedSerializationCoder(ordV)))
+              .apply(KVToTuple[K, V])
+          }
+        },
+        kryoCoder)
+    }
+
+    def mapSideAggregator(size: Int, semigroup: Semigroup[V])(implicit kryoCoder: KryoCoder): BeamOp[(K, V)] = {
+      TransformBeamOp[(K, V), (K, V)](
+        op,
+        new PTransform[PCollection[(K, V)], PCollection[(K, V)]]() {
+          override def expand(input: PCollection[(K, V)]): PCollection[(K, V)] = {
+            input
+              .apply(TupleToKV[K, V](kryoCoder, kryoCoder))
+              .apply(ParDo.of(MapSideAggregator[K, V](size, semigroup)))
+              .apply(KVToTuple[K, V])
+          }
+        },
+        kryoCoder)
+    }
+  }
+
+  case class SortGroupedValues[K, V](implicit ordV: Ordering[V], kryoCoder: KryoCoder) extends PTransform[PCollection[KV[K, java.lang.Iterable[V]]], PCollection[KV[K, java.lang.Iterable[V]]]] {
+    override def expand(input: PCollection[KV[K, lang.Iterable[V]]]): PCollection[KV[K, lang.Iterable[V]]] = {
+      input
+        .apply(ParDo.of(MapFn[KV[K, java.lang.Iterable[V]], KV[K, java.lang.Iterable[KV[WrappedComparable[V], Unit]]]]{ elem =>
+          KV.of(elem.getKey, elem.getValue.asScala.map(x => KV.of(WrappedComparable(x, ordV), ())).asJava)
+        })).setCoder(KvCoder.of(kryoCoder, IterableCoder.of(KvCoder.of(WrappedAtomicCoder[V](ordV, OrderedSerializationCoder(ordV)), kryoCoder))))
+        .apply(SortValues.create[K, WrappedComparable[V], Unit](BufferedExternalSorter.options()))
```

Comment:
I don't think `SortValues` cares about comparability of values, so unfortunately we need to implement it ourselves...

Reply:
how did the test pass then? He has a test that sorts ints.

Reply:
ahh...

https://beam.apache.org/releases/javadoc/2.5.0/org/apache/beam/sdk/extensions/sorter/SortValues.html
```
This transform sorts by lexicographic comparison of the byte representations of the secondary keys and may write secondary key-value pairs to disk. In order to retrieve the byte representations it requires the input PCollection to use a KvCoder for its input, an IterableCoder for its input values and a KvCoder for its secondary key-value pairs.
```

yeah, so it sorts by the binary representation, which for integers is the same as the value sorting... which is how it passed.